src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -873,6 +873,127 @@
          :headers {"Content-Type" "application/octet-stream"}
          :body (get-code-content environment code-id file-path)}))))
 
+(defn is-bolt-project?
+  [dir]
+  (or (fs/exists? (str dir "/bolt-project.yaml"))
+      (fs/exists? (str dir "/bolt.yaml"))
+      (fs/exists? (str dir "/Boltdir"))))
+
+(defn find-bolt-project
+  [project-dir project-name]
+  (->> project-dir
+       io/file
+       .list
+       seq
+       (filter #(is-bolt-project? (str project-dir "/" %)))
+       (some #{project-name})))
+
+(defn get-project-root
+  [project-dir project-name]
```

Comment:
can `project-dir` be `bolt-project-dir`, to match what's used below? or maybe something like `bolt-projects-dir` might be clearer, to distinguish from what you've been calling `project-root`? but really it's the consistency I care about.